content/blog/2020-03-02-react-v16.13.0.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "React v16.13.0"
+author: [threepointone]
+---
+
+Today we are releasing React 16.13.0. It contains bugfixes and new deprecation warnings to help prepare for a future major release.
+
+### Warnings for some updates during render
+
+A React render function is supposed to be a pure function. We do support a local update during a render to [derive state from props](https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops). However, this only works on the same Component/Hook. If you call it during a render on a different component, this will now issue a warning.
+
+The fix is usually to move this state change into a `useEffect(...)`.
+
+### Warnings for some deprecated string refs
+
+[String Refs is an old legacy API](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs) which is discouraged and is going to be deprecated in the future. This release adds a new warning to some unusual edge cases that can't be fixed automatically. For example using a string ref in a render prop:
+
+```jsx
+class ClassWithRenderProp extends React.Component {
+  componentDidMount() {
+    something(this.refs.foo);
+  }
+  render() {
+    return this.props.children();
+  }
+}
+
+class ClassParent extends React.Component {
+  render() {
+    return (
+      <ClassWithRenderProp>{() => <View ref="foo" />}</ClassWithRenderProp>
+    );
+  }
+}
+```
+
+You most likely don't have code like this but if you do, we recommend that you convert it to [`React.createRef()`](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) instead:
+
+```jsx
+class ClassWithRenderProp extends React.Component {
+  foo = React.createRef();
+  componentDidMount() {
+    something(this.foo.current);
+  }
+  render() {
+    return this.props.children(this.foo);
+  }
+}
+
+class ClassParent extends React.Component {
+  render() {
+    return (
+      <ClassWithRenderProp>{foo => <View ref={foo} />}</ClassWithRenderProp>
+    );
+  }
+}
+```
+
+Fixing these edge cases will be a prerequisite to do before being able to apply an automatic codemod.
```

Comment:
Too much jargon IMO:

>Fixing these edge cases will be a prerequisite to do before being able to apply an automatic codemod.

Suggestion:

>In the future, we will provide an automated script (a "codemod") to migrate away from String Refs. We are enabling this warning first so that you can manually fix the few cases that the codemod can't handle.